fvm/evm/stdlib/contract.cdc

```diff
@@ -288,6 +290,59 @@ contract EVM {
                 value: value.attoflow
             ) as! Result
         }
+
+        /// Bridges the given NFT to the EVM environment, requiring a Provider from which to withdraw a fee to fulfill
+        /// the bridge request
+        access(all)
+        fun depositNFT(
+            nft: @NonFungibleToken.NFT,
+            feeProvider: &{FungibleToken.Provider}
+        ) {
+            EVM.borrowBridgeAccessor().depositNFT(nft: <-nft, to: self.address(), feeProvider: feeProvider)
+        }
+
+        /// Bridges the given NFT from the EVM environment, requiring a Provider from which to withdraw a fee to fulfill
+        /// the bridge request. Note: the caller should own the requested NFT in EVM
+        access(all)
+        fun withdrawNFT(
+            type: Type,
+            id: UInt256,
+            feeProvider: &{FungibleToken.Provider}
+        ): @NonFungibleToken.NFT {
+            return <- EVM.borrowBridgeAccessor().withdrawNFT(
+                caller: &self as &CadenceOwnedAccount,
+                type: type,
+                id: id,
+                feeProvider: feeProvider
+            )
+        }
```

Comment:
Not sure if we should allow withdraw for non-coa accounts? 
I can see we force the reference to be COA but why not just exposing it on the COA resource

Reply:
This issue here is that, the COA ultimately needs to execute a `safeTransfer` call to some recipient against some EVM contract - here an ERC721 - to escrow the EVM asset. Within the scope of the EVM contract, how do we know:
- the escrow recipient address?
- the EVM contract address to target the call against?
If we want to know these values in scope, we'd need to further engrain the bridge into the EVM contract and I feel that given the critical nature of the EVM contract it would be best to avoid that.

To decouple the circular dependency between the EVM contract and bridge logic contracts, I thought it best to  pass the COA reference on to an implementation of interfaces defined within this contract and handle the resolution of those values within a contract that more in scope of bridge configuration. The implementation this reference is getting passed to is defined [here](https://github.com/onflow/flow-evm-bridge/blob/main/cadence/contracts/bridge/FlowEVMBridgeAccessor.cdc#L46).

Reply:
I see, so basically you pass the reference to the COA, so the bridge can decide what calls to do.